File: ICPC/ICPC Central Russia Regional Contest - 2020/I_Pharaoh_hEx.py
import math

N = 100000
isPrime = [False] * (N + 5)
prime = []

def solve():
    n = int(input())
    m = n
    odd = []
    for i in prime:
        if m % i == 0:
            cnt = 0
            while m % i == 0:
                cnt += 1
                m //= i
            if cnt % 2 == 1:
                odd.append(i)
    if m != 1:
        odd.append(m)
    t = 1
    for x in odd:
        t *= x
    a = n // t
    b = int(math.sqrt(a)) + 1
    print(b * b * t - n)

if __name__ == "__main__":
    for i in range(2, int(math.sqrt(N)) + 1):
        if not isPrime[i]:
            for j in range(i * i, N + 1, i):
                isPrime[j] = True
    for i in range(2, N + 1):
        if not isPrime[i]:
            prime.append(i)
    # for i in range(10):
    #     print(prime[i], end=' ')
    test = 1
    # test = int(input())
    while test > 0:
        solve()
        test -= 1

File: ICPC/ICPC Central Russia Regional Contest - 2020/I_Pharaoh_hEx.cpp
#include <bits/stdc++.h>
#define ll long long
#define fi first
#define se second
#define pb push_back
const ll inf = 1e18;
const long double esp = 1e-12;
const ll mod = 1e9 + 7;
using namespace std;
using namespace __gnu_pbds;
#define ordered_set tree<pair<int, int>, null_type, less<pair<int, int>>, rb_tree_tag, tree_order_statistics_node_update>
const int N = 100000;
vector<bool> isPrime(N + 5, false);
vector<int> prime;
void solve()
{
    ll n, m;
    cin >> n;
    m = n;
    vector<int> odd;
    for (auto i : prime)
    {
        if (m % i == 0)
        {
            int cnt = 0;
            while (m % i == 0)
            {
                cnt++;
                m /= i;
            }
            if (cnt % 2 == 1)
            {
                odd.pb(i);
            }
        }
    }
    if (m != 1)
    {
        odd.pb(m);
    }
    ll t = 1;
    for (auto x : odd)
    {
        t *= x;
    }
    ll a = n / t;
    ll b = (ll)(sqrt(a)) + 1;
    cout << b * b * t - n << '\n';
}
int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    for (int i = 2; i * i <= N; i++)
    {
        if (!isPrime[i])
        {
            for (int j = i * i; j <= N; j += i)
                isPrime[j] = true;
        }
    }
    for (int i = 2; i <= N; i++)
    {
        if (!isPrime[i])
            prime.pb(i);
    }
    // for (int i=0; i<10; i++) cout << prime[i] <<' ';
    int test = 1;
    // cin>>test;
    while (test--)
    {
        solve();
    }
    return 0;
}